src/NServiceBus.RavenDB/Internal/TransactionRecoveryStorageVerifier.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace NServiceBus.RavenDB.Internal
+{
+    using Raven.Client;
+    using Raven.Client.Document;
+    using Raven.Client.Document.DTC;
+
+    static class TransactionRecoveryStorageVerifier
+    {
+        public static void ReplaceStorageIfNotSetByUser(IDocumentStore store)
+        {
+            var docStore = store as DocumentStore;
+
+            if (docStore == null)
+                return;
+
+            var currentStorage = docStore.TransactionRecoveryStorage;
+
+            // VolatileOnlyTransactionRecoveryStorage is the default. If set to anything else, don't change it!
+            if (currentStorage == null || currentStorage is VolatileOnlyTransactionRecoveryStorage)
```

Comment:
This is just the 3.0.x fix that didn't go so far as to do things *right*, because that would be a behavior change. This PR is bringing that fix into 4.x as is, but will eventually be overwritten by the 3.1 fix I'm doing in #143 where we do it the right way. So kind of an academic discussion for this PR.